oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDBConnection.java

```diff
@@ -61,16 +64,19 @@ final class MongoDBConnection {
     static MongoDBConnection newMongoDBConnection(@NotNull String uri,
                                                   @NotNull String name,
                                                   @NotNull MongoClock clock,
-                                                  int socketTimeout,
-                                                  boolean socketKeepAlive) {
+                                                  int socketTimeout) {
         CompositeServerMonitorListener serverMonitorListener = new CompositeServerMonitorListener();
-        MongoClientOptions.Builder options = MongoConnection.getDefaultBuilder();
-        options.addServerMonitorListener(serverMonitorListener);
-        options.socketKeepAlive(socketKeepAlive);
+        MongoClientSettings.Builder options = MongoConnection.getDefaultBuilder();
+        options.applyToServerSettings((builder) -> {
```

Comment:
No need for parentheses and curly braces for one line inside the lambda.
```suggestion
        options.applyToServerSettings(builder -> builder.addServerMonitorListener(serverMonitorListener));
```